specification.md

```diff
@@ -1960,7 +1976,7 @@ If events defined in event-based conditions do not arrive before the states `eve
 | Parameter | Description | Type | Required |
 | --- | --- | --- | --- |
 | name | Data condition name | string | no |
-| condition | jq expression evaluated against state data. Must evaluate to true or false | string | yes |
+| [condition](#Workflow-Expressions) | Workflow expression evaluated against state data. Must evaluate to true or false | string | yes |
```

Comment:
For conditions, we need to have bindings on expression languages, meaning we need to specify how to use them. Any third-party extension is fine, but we need to say for jq that we want the result of the expression to be either JSON `true` or JSON `false` and that nil causes a runtime error.

Reply:
also feel free to postpone this in an issue

Reply:
jq can return true/false i would assume that anything else than true could be interpreted as false. so imo null/nil would be false in evaluation. i think other types of exceptions during expression evaluation should also just be handled as false in this case

Reply:
According to the JQ docs:

> false and null are considered "false values", and anything else is a "true value".

It seems like this would be really confusing for users of the workflow language to have to bridge that gap.  For example, they'd have to take extra special care for a pretty common/basic scenario like the following:

Input data:
```
{ "aNullableAgeField": null }
```

Given input like the above, a workflow might want to do something special if the user is a minor (or didn't provide their age).  The natural way to specify this in JQ would be:
```
"functions": [
  {
    "name": "isMinorOrNoAgeProvided",
    "operation": "${.aNullableAgeField < 18}",
    "type": "expression"
  }
]
```

Unfortunately that would return null for the sample input above, which is suggested to cause a runtime exception.

Workflow users would have to add a lot of boiler-plate to their expressions to guard against this, like so:
```
"functions": [
  {
    "name": "isMinorOrNoAgeProvided",
    "operation": "${.aNullableAgeField == null OR .aNullableAgeField < 18}",
    "type": "expression"
  }
]
```

Reply:
what would be the suggestion to fix this?

Reply:
Since we're targeting support for many expression lang choices, perhaps we can just mandate that the runtime is responsible for translating any potential value to a concrete true or false based on the standard behavior of that expression lang?   For the JQ example, this means that the runtime should translate according to JQ behavior?  Basically, we set the expectation that for any boolean expression, it's important that the runtime ensures that they properly translate any potential value from the user-defined expression to return exactly true or false and never any other value?

Reply:
+1
@jorgenj could you rewrite the current text in the spec to express what you stated here please?